src/main/scala/com/target/data_validator/ValidatorTable.scala

```diff
@@ -210,7 +211,18 @@ case class ValidatorHiveTable(
 
   override def getDF(session: SparkSession): Try[DataFrame] = {
     logger.info(s"Opening table: $db.$table")
-    Try(session.table(s"$db.$table"))
+    useHWC match {
+      case Some(x) if x =>
+        val hive = com.hortonworks.hwc.HiveWarehouseSession.session(session).build()
+        val query = condition match {
+          case Some(c) => s"SELECT * FROM $db.$table WHERE $c"
```

Comment:
Is it possible to rework code such that we can push the required column selections down to the HWC in this query?

As it stands this will take no advantage of columnar storage if you are only inspecting one column on an extremely wide table.